### PR TITLE
Switch CI to Ubuntu 20.04, add clang-10 build

### PR DESCRIPTION
**Pull request by mrks**
* refs #2074
* Adds a ubuntu/clang-10 build in CI but sticks with clang-9 as the default compiler. Mac is unchanged.
* Adds `cmake -DHYRISE_RELAXED_BUILD=On ..`, which disables most warnings.
* I verified that all scripts still work for Ubuntu 19.04, so while it is not our preferred system anymore, we don't break anything.

### CODE REVIEW

**Comment by mrks** [comment]
Interestingly, the clang-10 debug build is much slower...

File: resources/test_data/sqlite_testrunner_queries.sql [hunk 1]
```diff
@@ -287,7 +287,7 @@ sELEcT Sum(b + b) AS sum_b_b from mixed;
 
 -- Aggregates with NULL
 SELECT a, MAX(b) FROM mixed_null GROUP BY a;
-SELECT a, MAX(b) FROM mixed_null GROUP BY a ORDER BY MAX(b) DESC;
+SELECT a, MAX(b) FROM mixed_null GROUP BY a ORDER BY MAX(b), a DESC;
```
Comment (Bensk1) [inline resources/test_data/sqlite_testrunner_queries.sql:290]: Hm?

**Comment by Bensk1** [comment]
> Interestingly, the clang-10 debug build is much slower...

It takes roughly double the time of clang-9. Any idea at what point we are going to switch to clang-10?

**Comment by Bensk1** [comment]
> * I verified that all scripts still work for Ubuntu 19.04, so while it is not our preferred system anymore, we don't break anything.

Nice, we have never done that before :D

File: resources/test_data/sqlite_testrunner_queries.sql (hunk 1 above)
Comment (mrks) [reply resources/test_data/sqlite_testrunner_queries.sql:290]: If the query contains "ORDER BY", we expect the results to match line-by-line. However, the output order of GROUP BY is undefined, and the new libc++ changed it.

**Comment by mrks** [comment]
> It takes roughly double the time of clang-9. Any idea at what point we are going to switch to clang-10?

No idea. We should have a good reason to.

File: install_dependencies.sh [hunk 2]
```diff
@@ -53,11 +53,18 @@ if echo $REPLY | grep -E '^[Yy]$' > /dev/null; then
         if [ -f /etc/lsb-release ] && cat /etc/lsb-release | grep DISTRIB_ID | grep Ubuntu >/dev/null; then
             echo "Installing dependencies (this may take a while)..."
             if sudo apt-get update >/dev/null; then
+                sudo apt-get install --no-install-recommends -y software-properties-common lsb-release
```
Comment (Bensk1) [inline install_dependencies.sh:56]: Should we add lsb-release to the dependencies.md for the sake of completeness?

Comment (mrks) [reply install_dependencies.sh:56]: Thought about it but decided against it. We are only using it internally in install_dependencies.sh. If you look into dependencies.md, you probably cannot or don't want to run the install script and are looking for what Hyrise needs. lsb-release is not needed for that.

**Pull request merged**